lib/Parse/ParseDecl.cpp

```diff
@@ -2484,6 +2484,23 @@ parseLifetimeDescriptor(Parser &P,
   case tok::identifier: {
     Identifier name;
     auto loc = P.consumeIdentifier(name, /*diagnoseDollarPrefix=*/false);
+    
+    // In SIL, lifetimes explicitly state whether they are dependent on a
+    // memory location in addition to the value stored at that location.
+    if (P.isInSILMode()
+        && name.str() == "address"
+        && P.Tok.is(tok::integer_literal)) {
```

Comment:
Sorry, I glossed over this in my initial "review". I don't think the parser should have any logic that ties addressable parameters to lifetime descriptors. A lifetime descriptor is an element of the syntax. It doesn't tell you whether a lifetime dependency exists.

Reply:
This is purely there to allow the `LifetimeDependenceInfo` to be parsed back from canonical SIL. The `LifetimeDependenceInfo` still contains the canonical info in the end.

Reply:
Oh thanks. This `isSILMode` is a pretty big clue there